### PR TITLE
Add support for selecting process's subtree by middle clicking it

### PR DESCRIPTION
**Pull request by TheEragon**
This should fix the feature request #938

### CODE REVIEW

File: ProcessHacker/proctree.c [hunk 1]
```diff
@@ -4068,6 +4068,30 @@ BOOLEAN NTAPI PhpProcessTreeNewCallback(
...
+        if (ProcessTreeListSortOrder != NoSortOrder)
+            break;
+
+        PPH_TREENEW_MOUSE_EVENT mouseEvent = Parameter1;
+        if (!mouseEvent)
+            break;
+
+        node = (PPH_PROCESS_NODE)mouseEvent->Node;
+
+        // init last index to self, this way we select only the process if there are no children
```
Comment (dmex) [inline ProcessHacker/proctree.c:4083]: please add `(TheEragon)` at the end of these comments so we know who wrote them :P

Comment (TheEragon) [reply ProcessHacker/proctree.c:4083]: Will do :)

**Comment by TheEragon** [comment]
I just realized that this doesn't work like one might except, right now it simply toggles subtrees that you middle click on. But it should probably deselect already selected items and select just current subtree. CTRL+middle click should add subtree to already selected processes, this way it will be consistent with left clicking, what do you think?

**Comment by dmex** [comment]
> should probably deselect already selected items

Actions will apply to those items unless they're deselected.

**Review by dmex — APPROVED** [review]

**Comment by dmex** [comment]
👍

**Pull request merged**